dali/python/nvidia/dali/external_source.py

```diff
@@ -155,81 +155,99 @@ class ExternalSource():
 ----
 
 `source` : callable or iterable
-    The source of the data. The source is polled for data (via a call ``source()`` or ``next(source)``
-    whenever the pipeline needs input for the next iteration. The source can supply one or more data
-    batches, depending on the value of ``num_outputs``. If ``num_outputs`` is not set, the ``source`` is
-    expected to return a single batch. If it's specified, the data is expected to a be tuple or list
-    where each element corresponds to respective return value of the external_source.
-    If the source is a callable and has a positional argument, it is assumed to be the current
-    iteration number and consecutive calls will be ``source(0)``, ``source(1)``, etc.
-    If the source is a generator function, it is invoked and treated as an iterable - however,
-    unlike a generator, it can be used with ``cycle``, in which case the function will be called
-    again when the generator reaches end of iteration.
-    In the case of the GPU input, it is the user responsibility to modify the
-    provided GPU memory content only using provided stream (DALI schedules a copy on it
-    and all work is properly queued). If no stream is provided, DALI will use a default, with
-    best-effort approach at correctness (see ``cuda_stream`` argument documentation for details).
+    The source of the data.
+
+    The source is polled for data (via a call ``source()`` or ``next(source)``
+    when the pipeline needs input for the next iteration. Depending on the value of ``num_outputs``,
+    the source can supply one or more data batches. If ``num_outputs`` is not set, the ``source``
+    is expected to return one batch. If this value is specified, the data is expected to a be tuple,
+    or list, where each element corresponds to respective return value of the external_source.
+    If the source is a callable that accepts a positional argument, it is assumed to be the current
+    iteration number and consecutive calls will be ``source(0)``, ``source(1)``, and so on.
+    If the source is a generator function, the function is invoked and treated as an iterable.
+    However, unlike a generator, the function can be used with ``cycle``. In this case, the function
+    will be called again when the generator reaches the end of iteration.
+
+    For the GPU input, it is a user's responsibility to modify the provided GPU memory content
+    by using the provided stream. DALI schedules a copy on the stream, and all work is properly
```

Comment:
```suggestion
    only in the provided stream. DALI schedules a copy on this stream, and all work is properly
```
definitely not "by using".

Reply:
Done